src/buf_writer.c

```diff
@@ -5,8 +5,43 @@
 #include "jemalloc/internal/buf_writer.h"
 #include "jemalloc/internal/malloc_io.h"
 
+static void *
+buf_writer_allocate_internal_buf(size_t buf_len) {
+	if (buf_len > SC_LARGE_MAXCLASS) {
+		return NULL;
+	}
+	return iallocztm(TSDN_NULL, buf_len, sz_size2index(buf_len), false,
```

Comment:
Let’s avoid TSDN NULL unless we have to. If it’s only init and finish, passing in tsd should work?

Reply:
How about `tsdn`? In stats printing, we may not have a `tsd`.

Reply:
Yes sounds good.